### PR TITLE
Support creating Bool constants in the pattern_utils

### PR DESCRIPTION
**Pull request by mbrookhart**
@altanh Found a bug!

@gussmith23 @ZihengJiang Tagging you because git blame tells me you worked on the TVM_DTYPE_DISPATCH macro/MakeConstantScalar utility

### CODE REVIEW

**Comment by altanh** [comment]
any way to get this merged? LGTM (if that matters lol), pretty simple change

**Review by tqchen — APPROVED** [review]

**Pull request merged**
